src/python/pants/goal/anonymous_telemetry.py

```diff
@@ -0,0 +1,184 @@
+            )
+
+        if self._anonymous_telemetry.enabled:
+            repo_id = self._anonymous_telemetry.repo_id
+            if repo_id is None:
+                logger.error(
+                    f'Please set `repo_id = "<uuid>"` in the [anonymous-telemetry] section '
+                    f"of pants.toml, where `<uuid>` is some fixed random identifier, such as "
+                    f"one generated by uuidgen. No telemetry sent for this run. "
+                    f"{_telemetry_docs_referral}."
+                )
+            elif self.validate_repo_id(repo_id):
+                # Assemble and send the telemetry.
+                # Note that this method is called with finished=True only after the
+                # StreamingWorkunitHandler.session() ends, i.e., after end_run() has been called,
+                # so the RunTracker will have had a chance to finalize its state.
+                telemetry_data = context.run_tracker.get_anonymous_telemetry_data(repo_id)
+                # TODO: Add information about any errors that occurred.
+
+                reporter = Reporter(
+                    name="pantsbuild/pants",
+                    # We've already established consent at this point.
+                    consent=HumbugConsent(True),
+                    session_id=telemetry_data.get("run_id", str(uuid.uuid4())),
+                    bugout_token=_bugout_access_token,
+                    bugout_journal_id=_bugout_journal_id,
+                    timeout_seconds=5,
+                )
+
+                # This is copied from humbug code, to ensure that future changes to humbug
+                # don't add tags that inadvertently violate our anonymity promise.
+                system_tags = [
+                    "humbug",
+                    "source:{}".format(reporter.name),
+                    "os:{}".format(reporter.system_information.os),
+                    "arch:{}".format(reporter.system_information.machine),
+                    "python:{}".format(reporter.system_information.python_version_major),
+                    "python:{}.{}".format(
+                        reporter.system_information.python_version_major,
+                        reporter.system_information.python_version_minor,
+                    ),
+                    "python:{}".format(reporter.system_information.python_version),
+                    "session:{}".format(reporter.session_id),
+                ]
+                tags = (
+                    system_tags
+                    + [
+                        f"pants_version:{telemetry_data.get('pants_version')}",
+                    ]
+                    + [f"goal:{goal}" for goal in telemetry_data.get("goals", [])]
+                )
+
+                report = Report(
+                    title=f"pants run {reporter.session_id}",
+                    tags=tags,
+                    content=json.dumps(telemetry_data, sort_keys=True),
+                )
```

Comment:
FWIW, this library will create a thread there... which I am not sure is something we want happening from a rule.
https://github.com/bugout-dev/humbug/blob/d57f46fea72dbf2a5009765d2865b7ef1c153c38/python/humbug/report.py#L59

Reply:
@asherf : Good point, and completely reasonable that you would want to handle this kind of behavior yourself.

Just published `v0.1.9` of the library, in which you can instantiate a reporter in synchronous mode so that no thread is created: https://github.com/bugout-dev/humbug/blob/python/v0.1.9/python/humbug/report.py#L49

Reply:
Thanks @zomglings! Benjy, please do use that synchronous version. This is important for Pants to handle, and it will already use a background thread when `--pantsd` is enabled.

Reply:
Fixed.